app/src/main/java/com/example/levi9application/viewModel/CocktailViewModel.kt

```diff
@@ -1,26 +1,33 @@
 package com.example.levi9application.viewModel
 
+import android.app.Application
+import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
 import com.example.levi9application.model.Cocktail
+import com.example.levi9application.model.CocktailDatabase
 import com.example.levi9application.model.Resource
+import com.example.levi9application.repositories.CocktailDataRepo
 import com.example.levi9application.repositories.CocktailRepo
 import dagger.hilt.android.lifecycle.HiltViewModel
 import kotlinx.coroutines.CoroutineExceptionHandler
+import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.Job
 import kotlinx.coroutines.delay
 import kotlinx.coroutines.launch
 import javax.inject.Inject
 
 @HiltViewModel
 class CocktailViewModel
-@Inject constructor(private val cocktailRepo: CocktailRepo) : ViewModel() {
+@Inject constructor(private val cocktailRepo: CocktailRepo, application: Application) :
```

Comment:
We should not inject Application class here. The reason is that we should not use DAO inside the view model. Dao should be injected inside a Repo class. And Repo classes should be injected in the VM. Follow the pattern: 
Fragment -> VM -> Repo -> Dao -> DB.